addon_toolkit/interfaces/storage.py

```diff
@@ -152,18 +154,23 @@ class StorageAddonImp(AddonImp):
     config: StorageConfig
 
 
-@dataclasses.dataclass(frozen=True)
-class HttpRequestorImp:
+@dataclasses.dataclass
+class StorageAddonHttpRequestorImp(StorageAddonImp):
     """base class for storage addon implementations using GV network"""
 
     network: HttpRequestor
 
 
-@dataclasses.dataclass()
-class ClientRequestorImp:
+@dataclasses.dataclass
+class StorageAddonClientRequestorImp(StorageAddonImp):
     """base class for storage addon with custom clients"""
 
-    client: typing.Any
+    client: typing.Any = dataclasses.field(init=False)
+    credentials: dataclasses.InitVar[Credentials]
+
+    def __post_init__(self, credentials):
+        self.client = self.create_client(credentials)
 
-    def construct_client(self, account):
-        pass
+    @abstractmethod
```

Comment:
Does this decorator do anything without the the ABCMeta metaclass? My gut says we should only need either the `@abstractmethod` decorator or the `NotImplementedError`